index.html

```diff
@@ -1338,6 +1336,53 @@ <h3>alsoKnownAs</h3>
         </div>
 
       </section>
+
+      <section>
+        <h3>deterministicSameAs</h3>
+          <p>
+            The <code><a>deterministicSameAs</a></code> property is an array 
+            of one or more DID URI strings that are verified by the <a>DID Method</a> 
+            to be deterministically equivalent variants of the DID in the 
+            <code>id</code> property. The resolving party MUST treat the DIDs in the 
+            <code>id</code> property and <code><a>deterministicSameAs</a></code> array 
+            as references to the same logical ID, thus the resolving party MUST retain 
+            the DID URIs from the <code>id</code> and <code><a>deterministicSameAs</a></code> 
+            properties to ensure any subsequent interactions with them are correctly 
+            handled in relation to the <a>DID Subject</a>. (e.g. retain all variants in 
+            a database relationship so a login with any one of them maps back to the 
+            same UUID row for the <a>DID Subject</a>)
+          </p>
+  
+          <p>
+  <a>DID documents</a> MAY include the <code><a>deterministicSameAs</a></code> property.
+          </p>
+  
+          <dl>
+            <dt><dfn>deterministicSameAs</dfn></dt>
```

Comment:
This property isn't about the controller claiming two separate IDs, which need to bi-directionally confirm each other, are controlled by the same entity, it's one or more identifiers that are deterministic variants/aliases of the identifier that was resolved, which don't require any such bi-directional willing/subjective confirmation, because they're just "manifestations of the same thing*. There's literally 0 security vulnerability, in fact, to the contrary: the only values the field can be populated with are deterministic variants/aliases of the identifier that was resolved, which the method can guarantee are correct.

Reply:
> ...which the method can guarantee are correct.

Well, this is tricky. I think we may need to give more advice to DID Method implementers/producers here. Consider the case where a DID Method, `foo`, that allows arbitrary properties in a DID Document but does not do anything special with `deterministicSameAs` -- thereby allowing DID controllers to set it (I can imagine something like `did:web` doing this). Now applications must be careful not to use this value when the method in the DID in the `id` is `foo`. That's where a potential security vulnerability may lie. This implies we need to tell DID methods that they MUST do something about this field when they see it (either disallow it or validate it). It isn't enough for us to talk about the consuming side.

We have a passive statement above:

> ...of one or more DID URI strings that are verified by the <a>DID Method</a>...

But it seems that this needs to be transformed (or additional language added) into some kind of normative requirement on DID methods to disallow/validate the values in the field -- because it can't be ignored by them. Alternatively, we could allow it to be ignored, but this would impose additional burden on application developers. Applications today are written against resolvers that they trust (that will resolve against some number of pre-approved/pre-trusted DID methods). If we allow DID methods to split on ensuring the values in this field are verified, then the applications will need to do extra work to avoid a security vulnerability as noted in the above paragraph.

Reply:
@dlongley I agree with your point, however are there not other fields like this too e.g [controller](https://w3c.github.io/did-core/#control) that methods like `did:web` would be required to interpret and enforce proper usage?

Reply:
> thereby allowing DID controllers to set it (I can imagine something like did:web doing this

I can imagine this too, however I think that it is symptomatic of the overly simplified did:web implementations that exist today. In reality a did:web resolver will need to be slightly more than just an HTTP client, e.g there might be a schema validation layer that would handle the usage of properties like controller or deterministicSameAs